crates/composition/src/compose_supergraph.rs

```diff
@@ -0,0 +1,52 @@
+use crate::{
+    subgraphs::{DefinitionKind, DefinitionWalker},
+    Context,
+};
+
+pub(crate) fn build_supergraph(ctx: &mut Context<'_>) {
+    ctx.subgraphs
+        .iter_definition_groups(|first, rest| match first.kind() {
+            DefinitionKind::Object => merge_object_definitions(ctx, first, rest),
+            _ => todo!(),
+        });
+
+    ctx.subgraphs.iter_field_groups(|first, rest| {
+        if let Some(next) = rest.next() {
+            ctx.diagnostics.push_fatal(format!(
+                "The field `{}` on `{}` is defined in two subgraphs (`{}` and `{}`).",
```

Comment:
I don't fully understand this yet, but could there be three subgraphs defining the same field, would this error then be wrong?

Reply:
You would get the error for the first two, so it's not strictly wrong, but it's a confusing error, I agree. We can make the error better by mentioning all the subgraphs where the object type defines the same field. And we should add "without `@shareable`" but we're not at that level of validation yet.